test/worker/test-worker.ts

```diff
@@ -0,0 +1,41 @@
+import test from "ava";
+import * as sinon from "sinon";
+import { WorkerWrapperImpl } from "@worker/worker";
+import Storage from "@worker/storage";
+import { StoredWordList } from "@worker/storage";
+import * as compiler from "@predictive-text-studio/lexical-model-compiler";
+
+test("compile model should throws error when no file is found in the IndexedDB", async (t) => {
+  const storageStub = new Storage();
+  sinon.stub(storageStub, "fetchAllFiles").returns(Promise.resolve([]));
+
+  const workerWrapper = new WorkerWrapperImpl(storageStub);
+
+  const error = await t.throwsAsync(workerWrapper.compileModel());
+  t.is(error.message, "Cannot find any file in the IndexedDB");
+});
+
+test("compile model", async (t) => {
+  const testStoredFile = {
+    id: 1,
+    name: "test",
+    wordlist: [["test", 1]],
+  } as StoredWordList;
+
+  const storageStub = new Storage();
+  sinon
+    .stub(storageStub, "fetchAllFiles")
+    .returns(Promise.resolve([testStoredFile]));
+  const compilerMock = sinon.mock(compiler);
+  compilerMock
+    .expects("WordListFromArray")
+    .once()
+    .withArgs(testStoredFile.name, testStoredFile.wordlist);
+  compilerMock.expects("compileModelFromLexicalModelSource").once();
```

Comment:
Honestly, I'm not sure what this is exactly trying to exercise. I would test behaviour rather than, did this call this thing I know it will call. 

I think I'd prefer assertions that treat the function as a black box and verify properties about the input and the expected output. Look at `test-storage` to see how a geniune `Storage` class is instantiated and used to test the underlying functionality.